tests/integration/assessment/test_workflows.py

```diff
@@ -1,8 +1,17 @@
 from datetime import timedelta
 
+from databricks.sdk import WorkspaceClient
 from databricks.sdk.errors import NotFound, InvalidParameterValue
 from databricks.sdk.retries import retried
 from databricks.sdk.service.iam import PermissionLevel
+from databricks.sdk.service.jobs import RunLifecycleStateV2State, TerminationCodeCode
+
+
+def _assert_run_success(ws: WorkspaceClient, run_id: int) -> None:
```

Comment:
I thought I remember that asserts within functions called from a pytest test do not bubble up correctly. You only see there is an assert not which failed

Reply:
That's a concern, because we do this in quite a few places. I just tested this with the following:
```python
def _assert_i_am_a_teapot() -> None:
    assert "I am a not a teapot" is not None
    assert not "I am not a teapot"

def test_nested_assert() -> None:
    """Check how an assertion failure is reported if it occurs in a function called from the test."""
    _assert_i_am_a_teapot()
```

When run from the IDE:
```
============================= test session starts ==============================
collecting ... collected 1 item

integration/assessment/test_workflows.py::test_nested_assert 

============================== 1 failed in 0.12s ===============================
FAILED      [100%]
tests/integration/assessment/test_workflows.py:20 (test_nested_assert)
def test_nested_assert() -> None:
        """Check how an assertion failure is reported if it occurs in a function called from the test."""
>       _assert_i_am_a_teapot()

integration/assessment/test_workflows.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _assert_i_am_a_teapot() -> None:
        assert "I am a not a teapot" is not None
>       assert not "I am not a teapot"
E       AssertionError: assert not 'I am not a teapot'

integration/assessment/test_workflows.py:19: AssertionError
```

I think that looks fine?